Discussion: record-elektra use cases (#4462)
ElektraInitiative/libelektra: doc/usecases/record_elektra/UC_interactive_configuration.md

# Use Case: Interactively creating system config

## Summary

- **Title:** Interactively creating system config
- **Scope:** Configuration

> - **Level:** User Goal

Reviewer: This must be somewhere else, otherwise "Level:" is useless.

> - **Actors:** User (usually sysadmin)

Reviewer: Be as specific as possible. Who do we target?

```suggestion
- **Actors:** System administrator
```

- **Brief:** In order to interactively create a workable system configuration, sometimes multiple iterations are required. 

> It should be possible to modify the configuration on a single host, and then apply the changes onto all other hosts.

Reviewer: Why limit to modifications on a single host? As long the changes can somehow be merged...

```suggestion
It should be possible to modify the configuration directly on hosts, and then apply the changes onto all other hosts.
```

Author: The problem is, if you change the configuration on multiple systems at the same time, we'd have to manually merge the playbooks - as in the user would have to copy all the playbooks from the machines and let a tool merge them. 

If we limit this to a single host at a time, the ansible export plugin could do this automatically at this host.

To integrate seamlessly into an Ansible workflow, it should also be possible to merge the changes into the original playbook.

## Scenarios

- **Precondition:** -
- **Main success scenario:** 
  - User connects to the host and changes settings in Elektra
  - User exports a playbook that contains either
    - only the changed keys
    - the original playbook with the current changes applied

> - **Alternative scenario:** If the host was not configured using ansible-elektra, it is only possible to export a playbook with the changed keys

Reviewer: ```suggestion
- **Alternative scenario:** If the host has no applied playbook, it is only possible to export a playbook with the changed keys
```

Author: Hmm we can't do it with every playbook. It must be a playbook that was specifically exported by our plugin to do this, as we need to copy the playbook to the remote hosts for the merging to work.

Reviewer: I did not intend to change the meaning in this way. You can simply ignore the suggestion.
